build-tools/Xamarin.Android.Tools.BootstrapTasks/packages.config

```diff
@@ -1,7 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<packages>
-  <package id="Microsoft.DotNet.ApiCompat" version="5.0.0-beta.19606.1" targetFramework="net472" />
```

Comment:
This removal is an odd one; see also:

  * https://github.com/xamarin/xamarin-android/pull/3884#discussion_r354643291
  * https://github.com/xamarin/xamarin-android/pull/3997

It's the `<CheckApiCompatibility/>` in `build-tools/Xamarin.Android.Tools.BootstrapTasks` task which actually uses the `Microsoft.DotNet.ApiCompat` package outputs, so it makes sense *to me* for `build-tools/Xamarin.Android.Tools.BootstrapTasks` to be responsible for restoring this package.

The alternative is that *every project* which uses `<CheckApiCompatibility/>` task would also need to have this `@(PackageReference)`.  (Which certainly could be done -- you do so here! -- but it just "feels weird" to me.)

I would *prefer* that `Microsoft.DotNet.ApiCompat` be added as a `@(PackageReference)` in `Xamarin.Android.Tools.BootstrapTasks.csproj`.

Reply:
I've moved this reference back and added a `globalPackagesFolder` path to our `NuGet.config` so that we can continue to use local paths for the tooling.

Reply:
The scope of change needed to introduce a new `globalPackagesFolder` is a bit larger than anticipated, we should do that in a separate PR. For this PR I've updated the path to the ApiCompat tooling in `Mono.Android.targets` to use the default user based folder.